splunk_eventgen/lib/eventgentimer.py

```diff
@@ -148,12 +153,14 @@ def real_run(self):
                         genPlugin.updateConfig(config=self.config, outqueue=self.outputQueue)
                         genPlugin.updateCounts(count=count, start_time=et, end_time=lt)
                         try:
-                            self.generatorQueue.put(genPlugin)
+                            self.generatorQueue.put(genPlugin, True, 3)
                             self.executions += 1
+                            backfillearliest = lt
                         except Full:
-                            self.logger.warning("Generator Queue Full. Skipping current generation.")
-                        backfillearliest = lt
-                        
+                            self.logger.warning("Generator Queue Full. Reput the backfill generator task later. %d backfill generators are dispatched.", self.executions)
+                            backfillearliest = et
```

Comment:
Are we generating events going backwards in time? Will the events be indexed in order?

Reply:
when the queue is full, it will wait until it has slot. So events will be generated and indexed in order  but not in time as restriction of the queue size. As discussed in last meeting, the queue size is configurable.